libraries/compute-utils/function-views/src/function-view.ts

```diff
@@ -14,46 +14,46 @@ export abstract class FunctionView extends DG.ViewBase {
   protected _funcCall?: DG.FuncCall;
   protected _lastCall?: DG.FuncCall;
   protected _type: string = 'function';
+  protected funcName: string;
 
   // emitted when after a new FuncCall is linked
-  protected funcCallReplaced = new Subject<true>();
-
-  // emitted when after an initial FuncCall is linked
-  public onFuncCallReady = new BehaviorSubject<false>(false);
+  public funcCallReplaced = new Subject<true>();
 
   /**
-   * Constructs a new view using function with the given {@link funcName}. An fully-specified name is expected.
+   * Constructs a new view using function with the given {@link func}. An fully-specified name is expected.
    * Search of the function is async, so async {@link init} function is used.
-   * All other functions are called only when initialization is over and {@link this.onFuncCallReady} is emitted.
-   * @param funcName Name of DG.Func (either script or package function) to use as view foundation
+   * All other functions are called only when initialization is over and {@link this.onFuncCallReady} is run.
+   * @param initValue Name of DG.Func (either script or package function) or DG.FuncCall to use as view foundation
    * @param options Configuration object for the view.
    */
   constructor(
-    protected funcName: string,
+    protected initValue: string | DG.FuncCall,
     public options: {historyEnabled: boolean, isTabbed: boolean} = {historyEnabled: true, isTabbed: false},
   ) {
     super();
     this.box = true;
+    if (initValue instanceof DG.FuncCall) {
```

Comment:
Let's shorten it:

`this.funcName = initValue instanceof DG.FuncCall ? initValue.func.name : initValue;`

Reply:
Even better, we can safely remove these lines. `funcName` is never used in the library and never used in third-paty code (just checked).